siflower: switch to kernel 6.12, drop support for 6.6

GoetzGoerisch opened the pull request.

GoetzGoerisch committed:
siflower: switch to kernel 6.12

switch the kernel to 6.12 by default

Signed-off-by: Goetz Goerisch <ggoerisch@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20893
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

GoetzGoerisch committed:
siflower: drop support for kernel 6.6

Drop support for kernel 6.6 as now kernel 6.12 is set as default kernel
version.

Signed-off-by: Goetz Goerisch <ggoerisch@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20893
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

udanfacy22 commented:
See @raenye's comment here for issues they found: https://forum.openwrt.org/t/kernel-6-12-testing-needed-at91-bcm47xx-bcm4908-bcm53xx-qoriq-siflower-and-zynq-targets/242857/28

namiltd commented:
@GoetzGoerisch I don't remember what it was about, so it was probably a mistake/out of date.

GoetzGoerisch committed:
siflower: add kmod-i2c-designware-platform to device packages

adding the kmod-i2c-designware-platform by user suggestion

Co-authored-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: Goetz Goerisch <ggoerisch@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20893
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

GoetzGoerisch committed:
siflower: refresh kernel-6.12 config

refreshed with make kernel_oldconfig CONFIG_TARGET=subtarget

Signed-off-by: Goetz Goerisch <ggoerisch@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20893
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

raenye commented:
> The sf218898's PCIe interface has a defect; it cannot recognize MTK network cards, but it can recognize Qualcomm's NFA765 

Same here. MT7925 not detected (PCIe-wise; bluetooth is USB and is detected) whereas Intel 7260AC (some random mini PCIe card I had) is fully detected.

This is not just official OpenWrt; same happens with SinoVoip's 5.10-based build.

GoetzGoerisch commented:
was 6.6 working in this regard?

GoetzGoerisch commented:
Thank you for pointing out, that there was a description missing.

What to do with rest?

libiunc commented:
LGTM
I am researching HWNAT for SF21.

Ansuel commented:
@libiunc can you test if the card worked on 6.6 ?

libiunc commented:
> i'm asking if this is a real regression compared to 6.6 or it never worked... please compare between 6.6 and 6.12 for the stuff that doesn't work on 6.12

The problem with 6.12 is the same as with 6.6. The switch's status lights and the PCIe slots on the back (at least that's what I know) are faulty.No new or reduced faults.

libiunc commented:
I initially suspected a power supply issue with the MiniPCIe(like BPI-R64 1.2), but I recently had an asthma flare-up and can't use a soldering iron.

Ansuel commented:
@libiunc please keep care... Those soldering fume are bad hope you use an extractor. I also do this mistake from time to time and got an headache every time.
